src/ray/raylet/node_manager.cc

```diff
@@ -916,7 +916,11 @@ void NodeManager::ResourceDeleted(const NodeID &node_id,
     }
     RAY_LOG(DEBUG) << "[ResourceDeleted] received callback from node id " << node_id
                    << " with deleted resources: " << oss.str()
-                   << ". Updating resource map.";
+                   << ". Updating resource map. skip=" << (node_id == self_node_id_);
+  }
+
+  if (node_id == self_node_id_) {
```

Comment:
only this line

Reply:
Just curious, what is the mechanism that processing resource deletion for this node slows down the test and makes it flaky?

Reply:
oh, actually it's not slowing down the test. it's hanging and timeout eventually.

Basically, the local resource was deleted before scheduling and raylet found no resource to run the job.